src/FieldWarning/Assets/Units/Resources/UnitConfigSchema.json

```diff
@@ -0,0 +1,147 @@
+                }
+            }
+        },
+
+        "Mobility": {
+            "type": "object",
+            "description": "Values for the unit's mobility type\n\nSee: <MobilityType>",
+            "additionalProperties": false,
+            "required": [
+                "SlopeSensitivity",
+                "DirectionalSlopeSensitivity",
+                "PlainSpeed",
+                "ForestSpeed",
+                "WaterSpeed"
+            ],
+            "properties": {
+                "SlopeSensitivity": {
+                    "type": "number",
+                    "description": "More all-terrain units like infantry should have reduced slope sensitivity"
+                },
+                "DirectionalSlopeSensitivity": {
+                    "type": "number",
+                    "description": "A value of 0.5 means the unit will go the same speed on flat terrain as it does on a 30 degree downhill incline"
+                },
+                "PlainSpeed": {
+                    "type": "number",
+                    "description": "TODO: Add description"
+                },
+                "ForestSpeed": {
+                    "type": "number",
+                    "description": "TODO: Add description"
+                },
+                "WaterSpeed": {
+                    "type": "number",
+                    "description": "TODO: Add description"
+                }
+            }
+        }
+    },
+
+    "title": "Unit Config Schema",
+    "type": "object",
+    "required": [
+        "$schema",
+        "ID",
+        "CategoryKey",
+        "Name",
+        "Price",
+        "PrefabPath",
+        "Data",
+        "Mobility"
+    ],
+    "additionalProperties": false,
+    "properties": {
+        "$schema": {
+            "type": "string"
+        },
+        "ID": {
+            "type": "string",
+            "description": "A unique ID for this unit.\n\nShould follow the format: `Unit__[category]--[model]`\ne.g.: `Unit__TNK--M1A2`",
+            "pattern": "^Unit__(HEL|INF|LOG|REC|SUP|TNK|VHC)--[-a-zA-Z0-9]+$"
+        },
+        "CategoryKey": {
+            "type": "string",
+            "description": "The category of this unit.\n\nSee: <PFW.Model.Armory.UnitCategory>",
+            "enum": [ "LOG", "SUP", "INF", "TNK", "REC", "VHC", "HEL" ]
+        },
+        "Name": {
+            "type": "string",
+            "description": "The full, human-friendly name of this unit.\n\nFor use in unit labels, the deck builder, etc."
+        },
+        "Price": {
+            "type": "number",
+            "description": "(float) The cost of purchasing this unit."
+        },
+        "PrefabPath": {
+            "description": "Path to the unit's prefab, relative to any Resources directory.\n\nNOTE: The prefab should not have any components besides the `Transform`, as all necessary components will be added programatically at runtime according to the properties specified in this config file.",
```

Comment:
This description actually isn't accurate anymore — I was trying to figure out a way to programatically add the Targeting/Turret components, and the JSON ended up being so convoluted I figured it would probably be easier to keep doing that in Unity on the prefabs themselves. Do we want to move that stuff to JSON as well?

Reply:
The main point of the json configuration is to ship those files with the game so players can edit them easily (modding). So we'd definitely want to expose the weapon data at some point. But I think having the configuration be partial is more than enough for now, you are correct that the turrets would be pretty convoluted and also the structure could change a bit when we adapt the unit code for mirror (networking).